py/miniconf-mqtt/miniconf/__init__.py

```diff
@@ -47,4 +47,4 @@ def handle_message(_client, topic, payload, _qos, _properties):
 
     await asyncio.sleep(discovery_timeout)
 
-    return discovered_devices
+    return list(set(discovered_devices))
```

Comment:
I would change the API and return a set (also start with a set instead of a list in `discovered_devices = set() ... discovered_devices.add(topic[:-len(suffix)]) ...`)